crates/task-impls/src/consensus/handlers.rs

```diff
@@ -720,6 +716,11 @@ pub async fn update_state_and_vote_if_able<TYPES: NodeType, I: NodeImplementatio
         return false;
     };
     drop(read_consnesus);
+
+    let Ok(version) = version(view, &vote_info.1.read().await.clone()) else {
+        error!("Failed to calculate the version.");
```

Comment:
I think it would be better/more accurate to just match on the result and if it's `Err(e)` print `e` (which should have a more detailed message)